### ISSUE TITLE
Add the possibility to directly mark a gap of a value after a different value

### ISSUE THREAD

**Issue by tudortimi** [wavedrom/wavedrom.github.io#19]
Currently, if I want to add a gap of "don't care" after some known value I need to write the following:

{name: 'addr', wave: 'x2x|', data: ['A0'] }

What this results in is:

![gap_current](https://cloud.githubusercontent.com/assets/8170909/17171193/d89fe6e8-53ef-11e6-9f6f-4dce7bcef8ce.PNG)

What I'd like to see, though is:

![gap_desired](https://cloud.githubusercontent.com/assets/8170909/17171199/e0fc4912-53ef-11e6-8013-afb5d1157296.png)

I don't think this is currently possible.

You have a paper at SNUG that mentions generating assertions from WJSON files. The SVA description for the two are different. The first case is:

addr == A0 #1 1 #[*] some_other_cond

whereas for the second case it is:

addr == A0 #[*] some_other_cond

In the first case the waves state that a "don't care" cycle is always present. Whereas in the second case, it's optional.

**Comment by drom** [comment]
@tudortimi here is related bug report: https://github.com/drom/wavedrom/issues/48

**Comment by drom** [comment]
We have new syntax for multi-cycle bricks that we can probably use:
```
{ signal: [
  { name: 'clk' },
  { name: 'addr', wave: 'x=<x|>=.', data: 'A0'}
]}
```
But it seems to have a bug :disappointed:
![wavedrom](https://cloud.githubusercontent.com/assets/511872/17197852/2f3f3b7c-5425-11e6-9912-a2ed15113a0e.png)

That makes the gap move right.

adding @itsayellow to discussion.

**Comment by itsayellow** [comment]
My workaround in situations like this is to use period: 0.5 and upsample the wave string.  Wavedrom does have 0.5 period resolution.  To upsample you can add a period after every character except the x before the break.
Like so:
{signal: [
  {name: 'addr', wave: 'x2x|', data: ['A0'] },
  {name: 'addr', wave: 'x.2.x|', period:0.5, data: ['A0'] }
]}
![wavedrom](https://cloud.githubusercontent.com/assets/12565784/17200823/5b134e60-543d-11e6-9877-db5a237fc31a.png)

**Comment by tudortimi** [comment]
If the "<...>" tag means that what's in there is only a single cycle, then this is exactly what I'd need.

**Comment by drom** [comment]
@tudortimi @itsayellow Yes, `<...>` is sub-cycle tag, giving one brick per character with any period.

I have bumped wavedrom and online editor with related fixes. Please try:
```
{ signal: [
  { name: 'clk',  wave: 'p.....'},
  { name: 'A1', wave: 'x=x|=x|=.', data: 'a1 b1' },
  { name: 'A2', wave: 'x=x|=x|=.', data: 'a2 b2', period: 2 },
  { name: 'A3', wave: 'x=x|=x|=.', data: 'a3 b3', period: 3 },
  { name: 'A4', wave: 'x=x|=x|=.', data: 'a4 b4', period: 4 },
  { name: 'B1', wave: 'x=<x|>=<x|>=.', data: 'a1 b1' },
  { name: 'B2', wave: 'x=<x|..>=<x|..>=.', data: 'a2 b2', period: 2 },
  { name: 'B3', wave: 'x=<x|....>=<x|....>=.', data: 'a3 b3', period: 3 },
  { name: 'B4', wave: 'x=<x|......>=<x|......>=.', data: 'a4 b4', period: 4 }
]}
```

**Comment by drom** [comment]
@tudortimi could you elaborate more on SVA idea above?

**Comment by tudortimi** [comment]
I haven't concretely thought about how to generate assertions from WJSON; I just wanted to use SVA to explain what I was interested in. If someone were to implement a backend to go from WJSON to SVA, the idea would be that gaps in the waveforms should be treated as delays in any assertions (i.e. cycles where there are no checks): ##[delay] <some_condition>.

**Comment by DHDOWNS** [comment]
Has anyone (so far as you know)  looked more in the SVA from WJSON yet? I've been playing around with the idea and I'm seriously considering doing something along those lines.
The main thing I'd need in WJSON is some way to indicate where to put the implication operators. I've been modifying the format with a vertical bar made of ':'. So this:
```
a_signal x...1:...
b_signal 0....:...
```
gives me the sequence (!b_signal && ($rose(a_signal)) and I follow that with the |-> implication, then grab everything else up to the next colon as a second sequence, etc. 
I prefix the signal names with "local_" to anything I want as a local variable, so I don't think I'd need any changes there. 
Let me know if you are aware of some other effort along these lines (I could take this off my list of interesting things to do and not even notice its absence).
